internal/pkg/correlation/models/event.go

```diff
@@ -0,0 +1,88 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package models
+
+import (
+	"encoding/json"
+
+	contract "github.com/edgexfoundry/edgex-go/pkg/models"
+)
+
+type Event struct {
```

Comment:
It is confusing that this has the same type name as edgex-go/pkg/models/Event. How about EventEnvelope or something?

Reply:
Well it's technically in another namespace so there isn't a name collision and it's got a different scope.

Reply:
Yes technically different types. We are coming from a readability point of view and ease to understand which type we are dealing with at a point in the code. Make sense?